api/prober/types.go

```diff
@@ -47,8 +47,8 @@ type Config struct {
 type DependentResourceInfo struct {
 	// Ref identifies a resource
 	Ref *autoscalingv1.CrossVersionObjectReference `json:"ref"`
-	// Optional should be true if this resource should be present. If the resource is optional then it should be false.
-	Optional *bool `json:"optional"`
+	// Optional should be false if this resource should be present. If the resource is optional then it should be true.
+	Optional bool `json:"optional,omitempty"`
```

Comment:
omitempty should not be there as this field is not really optional. If one does not specify a value a zero value for this will be assumed which will be false. docstring also needs to reflect that very clearly.

Reply:
updated the docstring